server/user_account/permissions.py

```diff
@@ -1,23 +1,29 @@
+from django.core.exceptions import ObjectDoesNotExist
 from rest_framework.permissions import BasePermission, IsAuthenticated
 from user_account.models import CustomUser
+from audit.permissions import IsAssignedSK
 
 
 class PermissionFactory:
     def __init__(self, request):
-        self.base_im_permissions = [IsAuthenticated, IsInventoryManager, HasSameOrgInBody]
         self.base_sa_permissions = [IsAuthenticated, IsSystemAdmin]
+        self.base_im_permissions = [IsAuthenticated, IsInventoryManager, HasSameOrgInBody]
+        self.base_sk_permissions = [IsAssignedSK]
         self.request = request
 
     def validate_is_sa(self):
-        return IsAuthenticated.has_permission(IsAuthenticated(), self.request, None) and \
-                IsSystemAdmin.has_permission(IsSystemAdmin(), self.request, None)
+        return IsSystemAdmin.has_permission(IsSystemAdmin(), self.request, None)
+
+    def validate_is_im(self):
+        return IsInventoryManager.has_permission(IsInventoryManager(), self.request, None)
 
-    def get_general_permissions(self, get_im_perms):
+    def get_general_permissions(self, additional_perms):
         if self.validate_is_sa():
             permission_classes = self.base_sa_permissions
+        elif self.validate_is_im():
+            permission_classes = self.base_im_permissions + additional_perms
         else:
-            # concatenate lists
-            permission_classes = self.base_im_permissions + get_im_perms
+            permission_classes = self.base_sk_permissions + additional_perms
         return permission_classes
```

Comment:
yes maybe the factory can/should accommodate SK users

Reply:
This was talked about over call, the way the factory is currently setup is meant only for SA and IMs. If We add the SKs into the factory as we did, yes we can accommodate general cases for the SK but the problem is now we have allowed for SKs to potentially gain access to other parts of our app which they wouldn't have otherwise. This isn't picked up since none of the tests test what we shouldn't have which is SK possibilities. We'll have to work on them as separate permissions unfortunatly

Reply:
yeah exactly, these changes can be made through a new subclass that can inherit some of the functionality but provide its own custom functionality.

Reply:
check the recent commits, I believe it addresses the security issue while still extending the functionality of the permission factory so it accommodates both sides